File: server/user-service/src/main/java/com/sudoo/userservice/service/UserService.java
package com.sudoo.userservice.service;

import com.sudo248.domain.exception.ApiException;
import com.sudoo.userservice.controller.dto.UserDto;
import com.sudoo.userservice.controller.dto.UserInfoDto;
import com.sudoo.userservice.repository.entitity.User;

public interface UserService {
    UserDto getUser(String userId);
    UserInfoDto getUserInfo(String userId) throws ApiException;
    UserDto postUser(UserDto userDto);
    UserDto putUser(String userId, UserDto userDto);
    UserDto toDto(User user);
    User toEntity(UserDto userDto);
}

File: server/product-service/src/main/kotlin/com/sudoo/productservice/service/UserService.kt
package com.sudoo.productservice.service

import com.sudoo.productservice.dto.AddressDto
import com.sudoo.productservice.dto.UserInfoDto

interface UserService {
    suspend fun getUserInfo(userId: String): UserInfoDto

    suspend fun postAddress(address: AddressDto): AddressDto

    suspend fun getAddressById(addressId: String): AddressDto
}